### ISSUE TITLE
AWS volume quota reached

### ISSUE THREAD

**Issue by justinsb** [kubernetes/test-infra#4361]
(copying from slack)

I think we have hit our AWS volume quota limit:

 ```PersistentVolume: VolumeLimitExceeded: You have exceeded your maximum gp2 storage limit of 20 TiB in this region. Please contact AWS Support to request an Elastic Block Store service limit increase.
```
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/kops/3331/pull-kops-e2e-kubernetes-aws/2849/?log#log

Each execution uses ~ 500 GB of disks though, so if we are running > 40 jobs concurrently it is possible we are not leaking.  (And we are seeing rate limit problems, which suggests each job is using too many requests or that we are running too many jobs in parallel) (edited)

If someone could have a look, and - presuming that there are older volumes - maybe send me the info on a few (in particular the size & tags, as that is our best clue...)

**Comment by justinsb** [comment]
Perhaps related to https://github.com/kubernetes/test-infra/pull/4332

**Comment by justinsb** [comment]
(Also https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/51121/pull-kubernetes-e2e-kops-aws/46672/?log#log and ongoing)

**Comment by liggitt** [comment]
> Each execution uses ~ 500 GB of disks

seriously?

**Comment by yastij** [comment]
@liggitt typo I think, I can't see why we would need so much disk per ececution

**Comment by justinsb** [comment]
It allocates EBS volumes of at least that much, which is what counts against quota (if need be we can reduce this, but I suspect we're either leaking or running too many concurrent jobs).

Most of that 500GB will remain unused though :-)

**Comment by chrislovecnm** [comment]
@justinsb are we hitting account limit or region limit?

**Comment by yastij** [comment]
@justinsb - we should definitly reduce it. if reduced we should be able to run the same number of jobs just fine ?

**Comment by justinsb** [comment]
@yastij can you share the volume tags you observed?

If we reduce it by a factor of 2, we can run twice as many jobs, but if we aren't limiting the number of jobs (cf #4322) that doesn't solve the root problem.  If on the other hand we are leaking volumes, we should try to plug the leak (cf  https://github.com/kubernetes/kubernetes/issues/50898) or look into why the clean job is broken (cf #4370)

**Comment by yastij** [comment]
@justinsb - To me, to unblock the situation at CI level, we should: 

1. reduce volume allocation
2. set a limit for concurrent jobs.

On the other hand, we should see if there's a leak, I thought that we couldn't have tags on EBS at creation time, but it seems that it's possible kubernetes/kubernetes#50898

**Comment by justinsb** [comment]
Are you able to look at the ebs volumes @yastij?

**Comment by yastij** [comment]
if it necessits rights to do so, I won't be able to take a look

**Comment by krzyzacy** [comment]
/assign @zmerlynn @justinsb 

I'll hand this to you - janitor should be back up now, feel free to closed if this is good, but we might still want to fix the actual leak

**Comment by chrislovecnm** [comment]
/remind-lifecycle stale

Keep this open with the new account. And with the new PR that I hacecwe may be able to increase number of runs using more regions

**Comment by chrislovecnm** [comment]
/remove-lifecycle stale

Let me try that again

**Comment by yastij** [comment]
@chrislovecnm - do we still have quota issues ?

**Comment by krzyzacy** [comment]
yes, we still need to configure quota for the new cncf-aws account

**Comment by yastij** [comment]
@krzyzacy - any PR open ?

**Comment by krzyzacy** [comment]
@yastij there's email thread with aws going on
https://github.com/kubernetes/test-infra/pull/5880 is the PR switches over blocking jobs to the new account

**Comment by chrislovecnm** [comment]
I also reduced the image volume size in another PR to help with this issue.  I think we can close this now as we have moved to the new account.  Thoughts?

**Comment by yastij** [comment]
I agree,  this is holding #6362
